hax/hax/util.py

```diff
@@ -1278,6 +1278,32 @@ def update_pver_bc(self, data: ByteCountStats) -> None:
             LOG.debug('Setting bytecount stats in KV: %s:%s', key, value)
             self.kv.kv_put(key, value)
 
+    def update_bc_for_dg_category(self,
+                                  pver_items: Dict[str, PverState]) -> None:
+        '''
+        This function will update bytecount for subsequent dg state/category
+        which will reflect on cluster status.
+        '''
+        pool_ver = self.kv.kv_get('ioservices/', recurse=True)
+        pver_state_map = {
+            PverState.M0_CPS_HEALTHY: 'healthy_byte_count',
+            PverState.M0_CPS_DEGRADED: 'degraded_byte_count',
+            PverState.M0_CPS_CRITICAL: 'critical_byte_count',
+            PverState.M0_CPS_DAMAGED: 'damaged_byte_count'
+        }
+
+        for pver, status in pver_items.items():
```

Comment:
I see status being used to updated the kv but not to calculated the bytecount as per the pver status. Also in your example all the bytecount is upated against `critical_bytecount` when threre are no failures, why?

Reply:
Yes, thanks for pointing it out. There was an issue with the algorithm, fixed it now. Please check.